### ISSUE TITLE
[UI] Save state slot doesn't reset to 0 when starting content

### ISSUE THREAD

**Issue by Brunnis** [libretro/RetroArch#12706]
## Description

The save state slot that is selected before choosing "Close content" becomes the same save state slot that will  be selected when launching new content. XMB and Ozone have been tested and behave the same.

### Expected behavior

I would expect the save state slot to reset to the default (0) when launching content.

### Actual behavior

Save state slot is remembered across content launches.

### Steps to reproduce the bug

1. Start any content and enter the Quick Menu.
2. Set save state slot to, let's say, 2.
3. Choose "Close content".
4. Launch any content and enter the Quick Menu. The save state slot is still 2 (or whatever you set in step 2 above).

### Version/Commit

- RetroArch: 1.9.6

### Environment information

- OS: Windows 10

**Comment by sonninnos** [comment]
Actually the selected slot `state_slot` is saved to global config for some reason, so there will be conflicts in resetting it if remembering is also desired.. I think it should go to zero on core unload or content close, but maybe someone knows better.

I hadn't noticed it at all since I don't have auto conf save enabled, so of course it is always 0 on launch.

**Comment by Brunnis** [comment]
@sonninnos Yeah, I noticed that it was saved to retroarch.cfg after I wrote this issue. Like you say, I assume there's probably some reason for this, although it's not obvious.

The auto conf save option, is that Settings -> Configuration -> Save Configuration on Quit you're referring to? I tried disabling it, but the save state slot is still remembered between content launches.

**Comment by sonninnos** [comment]
Sure, because it only prevents it from saving on quit, as the option name hints. That particular resource shouldn't get saved at all if you ask me.

**Comment by Brunnis** [comment]
Yeah, of course. From your previous message ("I hadn't noticed it at all since I don't have auto conf save enabled, so of course it is always 0 on launch."), it sounded like there was a setting that prevents automatic saving to the configuration file. The setting I mentioned was the closest I could find. Care to point me in the right direction?

**Comment by sonninnos** [comment]
That is the setting I meant. It either saves everything or nothing, so it is not a proper solution.

**Comment by Brunni132** [comment]
Too bad, because remembering the last state slot was also useful when we quit RetroArch and saved the same game…

See, I have a save_state button on my overlay. With auto state load enabled, so that I can quickly restart my games from a cold start. The state -1 will be loaded, but my save_state button will write the state 0, not the -1 as I want (I need, each time I start a game, to go in the Save states menu, and select the slot as auto -1).

I think what makes the most sense would be to remember the last state as previously, but PER GAME.

**Comment by vitran96** [comment]
I agree. I didnt think it was a bug at all until this was fix. So incremental save (up to 40) but not knowing my last save slot make me really confuse.

**Comment by Brunni132** [comment]
What should we do @vitran96 ? Reopen a ticket? Or can we ask to reopen this one?

**Comment by vitran96** [comment]
@Brunni132 I found this ticket #13062
This looks like it is related to what we want
